addons/website/static/src/builder/plugins/options/social_media_option_plugin.js

```diff
@@ -172,77 +162,76 @@ class SocialMediaOptionPlugin extends Plugin {
         ],
     };
 
-    /** The social media's name for which there is an entry in the orm */
-    async getRecordedSocialMediaNames() {
-        await this.fetchRecordedSocialMedia();
-        return this.recordedSocialMedia.keys();
-    }
-
-    // TODO: a method to give access to the `recordedSocialMedia` for facebook page and instagram page
+    /**
+     * Fetches social media links from the company record and pre-fills the
+     * corresponding links in the snippet, replacing internal
+     * "/website/social/<name>" entries.
+     */
+    async prefillSocialMediaLinks(snippetEl) {
+        this.companySocialLinks = new Map();
+        const socialMediaInfoFields = [
+            ...socialMediaInfo
+                .entries()
+                .filter(([name, info]) => info.recorded)
+                .map(([name, info]) => `social_${name}`),
+        ];
```

Comment:
`.filter().map()` takes 2 fullpasses and two new arrays used, meanwhile `for...of` if faster and memory efficient.

```suggestion
        const socialMediaInfoFields = [];
         for (const [name, info] of socialMediaInfo.entries()) {
           if (info.recorded) {
               socialMediaInfoFields.push(`social_${name}`);
           }
       }

```